### PR TITLE
completed the setup and cleared errors and finished with the add, lis…

### PR DESCRIPTION
**Pull request by sabarivijayan**
Completed the add,

### CODE REVIEW

**Comment by sabarivijayan** [comment]
completed project add, list, list by id, edit by id and delete by id with only minio config and upload left

File: src/config/db.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ const connectDB = async () => {
         const conn = await mongoose.connect(process.env.MONGO_URI)
         console.log(`MongoDB Connected: ${conn.connection.host}`)
         // Call the databaseMigration function to populate the database with initial data (optional)
-        databaseMigration()
+
```
Comment (karthikkrishnakumar) [inline src/config/db.js:13]: remove unwanted space

File: src/config/db.js [hunk 2]
```diff
@@ -10,7 +10,7 @@ const connectDB = async () => {
         const conn = await mongoose.connect(process.env.MONGO_URI)
         console.log(`MongoDB Connected: ${conn.connection.host}`)
         // Call the databaseMigration function to populate the database with initial data (optional)
-        databaseMigration()
+        
     } catch (error) {
         console.error(`Error: ${error.message}`)
```
Comment (karthikkrishnakumar) [inline src/config/db.js:15]: remove console

File: src/middlewares/authenticate-admin.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import jwt from 'jsonwebtoken'
-import EmployeeRepository from '../repositories/employeeRepository.js'
+// import EmployeeRepository from '../repositories/employeeRepository.js'
```
Comment (karthikkrishnakumar) [inline src/middlewares/authenticate-admin.js:2]: remove unwanted comments

File: src/middlewares/authenticate-admin.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
 
-const employeeRepo = new EmployeeRepository()
+// const employeeRepo = new EmployeeRepository()
```
Comment (karthikkrishnakumar) [inline src/middlewares/authenticate-admin.js:4]: remove unwanted comments

File: src/middlewares/authenticate-employee.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import jwt from 'jsonwebtoken'
-import EmployeeRepository from '../repositories/employeeRepository.js'
+// import EmployeeRepository from '../repositories/employeeRepository.js'
```
Comment (karthikkrishnakumar) [inline src/middlewares/authenticate-employee.js:2]: remove unwanted comments

File: src/middlewares/check-permission.js [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import Role from '../models/role.js'
+// import Role from '../models/role.js'
```
Comment (karthikkrishnakumar) [inline src/middlewares/check-permission.js:1]: remove unwanted comments

File: src/repositories/admin/project-repository.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
+import Project from "../../models/projects.js";
+
+export default class ProjectRepository {
+    /**
+     * Add a new project
+     * @param {Object} projectData - The project data
+     * @return {Promise<Project>} - The created project
+     */
+    async addProject(projectData) {
+        const project = new Project(projectData);
```
Comment (karthikkrishnakumar) [inline src/repositories/admin/project-repository.js:10]: include try catch and throw errors

File: src/repositories/admin/project-repository.js [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    /**
+     * Get all projects with pagination using POST
+     * @param {Object} options - Pagination options
+     * @param {Object} filters - Query filters from request body
+     * @return {Promise<Object>} - Paginated projects
+     */
+    async getAllProjects(options, filters = {}) {
+        const query = {};
```
Comment (karthikkrishnakumar) [inline src/repositories/admin/project-repository.js:21]: include try catch and throw errors

File: src/repositories/admin/project-repository.js [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    /**
+     * Get project by id
+     * @param {String} projectId - The project id
+     * @param {Object} filters - Additional filters from request body
+     * @return {Promise<Project>} - The project
+     */
+    async getProjectById(projectId, filters = {}) {
+        const query = { _id: projectId };
```
Comment (karthikkrishnakumar) [inline src/repositories/admin/project-repository.js:66]: include try catch and throw errors

File: src/repositories/admin/project-repository.js [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * Check if project name exists for the client
+     * @param {String} projectName 
+     * @param {String} clientName
+     * @param {String} excludeProjectId - Optional project id to exclude from check
+     * @return {Promise<Project>}
+     */
+    async checkProjectExists(projectName, clientName, excludeProjectId = null) {
+        const query = {
```
Comment (karthikkrishnakumar) [inline src/repositories/admin/project-repository.js:104]: include try catch and throw errors

File: src/requests/admin/add-project-request.js [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+      "any.required": "Please enter the project description.",
+    }),
+    plannedStartDate: Joi.required().messages({
+      "date.base": "Please enter a valid planned start date.",
+      "any.required": "Please enter the planned start date.",
+    }),
+    plannedEndDate: Joi.optional().allow("").allow(null),
+    actualStartDate: Joi.date().optional().allow("").allow(null),
+    actualEndDate: Joi.date().optional().allow("").allow(null),
+    // projectLead: Joi.string()
```
Comment (karthikkrishnakumar) [inline src/requests/admin/add-project-request.js:29]: remove unwanted comments

File: src/requests/admin/add-project-request.js [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    });
+
+    // Check if project exists
+    const checkProjectExists =
+      await AddProjectRequest.projectRepo.checkProjectExists(
+        this.data.projectName,
+        this.data.clientName
+      );
+
+    // Check if project lead exists
```
Comment (karthikkrishnakumar) [inline src/requests/admin/add-project-request.js:75]: remove unwanted comments

File: src/routes/admin.js [hunk 13]
```diff
@@ -1,12 +1,12 @@
 import express from 'express'
 import multer from 'multer'
 
-import {
-    checkAnyPermissions,
-    checkPermissions,
-} from '../middlewares/checkPermission.js'
-import { authenticateAdmin } from '../middlewares/authenticateEmployee.js'
-import uploadCsv from '../utils/uploadCsv.js'
+// import {
```
Comment (karthikkrishnakumar) [inline src/routes/admin.js:4]: remove unwanted comments

File: src/routes/employee.js [hunk 14]
```diff
@@ -14,7 +14,7 @@ const employeeRouter = express.Router()
 const multerMiddleware = multer().single('file')
 const upload = multer()
 
-const employee = new EmployeeController()
+// const employee = new EmployeeController()
```
Comment (karthikkrishnakumar) [inline src/routes/employee.js:17]: remove unwanted comments

File: src/utils/deleteFile.js [hunk 15]
```diff
@@ -1,12 +1,12 @@
 import fs from 'fs'
 import minioConfig from '../config/minio.js'
-import awsConfig from '../config/aws.js'
+// import awsConfig from '../config/aws.js'
```
Comment (karthikkrishnakumar) [inline src/utils/deleteFile.js:3]: remove unwanted comments

File: src/utils/uploadFile.js [hunk 16]
```diff
@@ -1,7 +1,7 @@
 import fs from 'fs'
 import { Readable } from 'stream'
 import minioConfig from '../config/minio.js'
-import awsConfig from '../config/aws.js'
+// import awsConfig from '../config/aws.js'
```
Comment (karthikkrishnakumar) [inline src/utils/uploadFile.js:4]: remove unwanted comments

**Review by karthikkrishnakumar — CHANGES_REQUESTED** [review]

File: src/controllers/admin/project-controller.js [hunk 17]
```diff
@@ -0,0 +1,458 @@
...
+      if (projectDetails) {
+        const projectData = await ProjectResponse.format(projectDetails);
+
+        res.status(200).json({
+          status: true,
+          message: "Project added successfully.",
+          data: projectData,
+        });
+      } else {
+        res.status(422).json({
```
Comment (karthikkrishnakumar) [inline src/controllers/admin/project-controller.js:103]: status code 200

File: src/controllers/admin/project-controller.js [hunk 18]
```diff
@@ -0,0 +1,458 @@
...
+          status: false,
+          message: "Failed to add project.",
+          data: [],
+        });
+      }
+    } catch (error) {
+      res.status(422).json({
```
Comment (karthikkrishnakumar) [inline src/controllers/admin/project-controller.js:110]: manage 500 also

**Review by karthikkrishnakumar — CHANGES_REQUESTED** [review]

**Pull request merged**